static/js/components/EducationForm.js

```diff
@@ -440,11 +435,26 @@ class EducationForm extends ProfileFormFields {
 
   render() {
     const {
-      ui: { showEducationDeleteDialog, educationDialogVisibility },
-      profilePatchStatus
+      ui: {
+        educationDialogIndex,
+        showEducationDeleteDialog,
+        educationDialogVisibility
+      },
+      profilePatchStatus,
+      profile
     } = this.props
 
     const inFlight = profilePatchStatus === FETCH_PROCESSING
+    const keySet = (key): any => ["education", educationDialogIndex, key]
+    const id = _.get(profile, keySet("id"))
+    const degreeName = _.get(
+      R.find(R.propEq("value", _.get(profile, keySet("degree_name"))))(
+        EDUCATION_LEVELS
+      ),
+      "label"
+    )
```

Comment:
This is a bit of a convoluted way to get this out here - why don't we instead declare an `EDUCATION_LEVEL_LABELS` object in the constants file, so that here you can just do `EDUCATION_LEVEL_LABELS[_.get(profile, keySet("degree_name"))]`?

I think it could be something like

```js
const EDUCATION_LEVEL_LABELS = Object.assign(
  {},
  EDUCATION_LEVELS.map(({value, label}) => [value, label])
)
```

you could also use `R.fromPairs` instead of `Object.assign`